src/order/bounded_order.lean

```diff
@@ -1101,6 +1101,13 @@ instance [has_le α] [order_top α] : order_bot (order_dual α) :=
 instance [has_le α] [bounded_order α] : bounded_order (order_dual α) :=
 { .. order_dual.order_top α, .. order_dual.order_bot α }
 
+open order_dual
+
+@[simp] lemma of_dual_bot [has_top α] : of_dual ⊥ = (⊤ : α) := rfl
+@[simp] lemma of_dual_top [has_bot α] : of_dual ⊤ = (⊥ : α) := rfl
+@[simp] lemma to_dual_bot [has_bot α] : to_dual (⊥ : α) = ⊤ := rfl
+@[simp] lemma to_dual_top [has_top α] : to_dual (⊤ : α) = ⊥ := rfl
```

Comment:
Should these be in the `order_dual` namespace?

Reply:
I don't think so. The lemma names are unambiguous, and `order_dual` as a namespace is *only* used for `to_dual` and `of_dual` (which are indeed ambiguous.

Reply:
My argument would be that anyone using `to_dual` and `of_dual` already has the namespace open anyway, so we should just put all these lemmas in the namespace.

Reply:
That's where you're wrong. About half the time, they don't open `order_dual`. `open .*order_dual` yields 33 files, 10 of which are not under `order.`. `order_dual.to_dual`  and `order_dual.of_dual` appear in 16 files, 8 of which are not under `order.`. Looking through the `open .*order_dual` results, half of them are because of me.

Reply:
Note that [`order_dual.to_dual_le_to_dual`](https://leanprover-community.github.io/mathlib_docs/order/order_dual.html#order_dual.to_dual_le_to_dual) is in the order_dual namespace, which suggests this is where the other lattice operations belong too.

Reply:
I'm with Eric on this - especially because of this last point about the existing lemmas like this being in that namespace

Reply:
Sample bias. The only lemmas that use the `order_dual` namespace are the ones in `order.order_dual` (~16 lemmas by being generous) and `order_dual.to_dual_smul`, `order_dual.of_dual_smul` (where there is indeed a risk of confusion because `smul` is an algebraic operation, not an order one). On the other hand we have [`is_max_to_dual_iff`](https://leanprover-community.github.io/mathlib_docs/order/max.html#is_max_to_dual_iff) and friends, [`is_bot.to_dual`](https://leanprover-community.github.io/mathlib_docs/order/max.html#is_bot.to_dual) and friends, [`of_dual_covby_of_dual_iff`](https://leanprover-community.github.io/mathlib_docs/order/cover.html#of_dual_covby_of_dual_iff), [`Icc_to_dual`](https://leanprover-community.github.io/mathlib_docs/order/locally_finite.html#Icc_to_dual) and friends, [`of_dual_compares_of_dual`](https://leanprover-community.github.io/mathlib_docs/order/compare.html#of_dual_compares_of_dual), [`grade_to_dual`](https://leanprover-community.github.io/mathlib_docs/order/grade.html#grade_to_dual), [`is_upper_set_preimage_of_dual_iff`](https://leanprover-community.github.io/mathlib_docs/order/upper_lower.html#is_upper_set_preimage_of_dual_iff)...

The namespace is used in one file and the norm everywhere else is to drop it.